lib/xnvme_be_ramdisk_sync.c

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) Mads Ynddal <m.ynddal@samsung.com>
+// SPDX-License-Identifier: Apache-2.0
+#ifndef _XOPEN_SOURCE
+#define _XOPEN_SOURCE 700
+#endif
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_RAMDISK_ENABLED
+#include <errno.h>
+#include <unistd.h>
+#include <libxnvme_spec_fs.h>
+#include <xnvme_dev.h>
+#include <xnvme_be_ramdisk.h>
+
+int
+xnvme_be_ramdisk_sync_cmd_io(struct xnvme_cmd_ctx *ctx, void *dbuf, size_t dbuf_nbytes,
+			     void *XNVME_UNUSED(mbuf), size_t XNVME_UNUSED(mbuf_nbytes))
+{
+	struct xnvme_be_ramdisk_state *state = (void *)ctx->dev->be.state;
+	const uint64_t ssw = ctx->dev->geo.ssw;
+
+	switch (ctx->cmd.common.opcode) {
+	case XNVME_SPEC_NVM_OPC_WRITE:
```

Comment:
You can add handling of the pseudo-fs-spec here, it is the same as for NVM but without the need to shift by ``ssw``.